jdaviz/configs/default/plugins/plot_options/plot_options.py

```diff
@@ -38,8 +46,19 @@ def _on_select_viewer_message(self, msg):
         # message from elsewhere requesting to change the selected viewer
         self.selected_viewer = msg.viewer
 
+    def _on_data_changed(self):
+        self.show_uncertainty = False
+
     @observe("selected_viewer")
     def _selected_viewer_changed(self, event={}):
         viewer = self.app.get_viewer_by_id(event.get('new', self.selected_viewer))
         self.viewer_widget = viewer.viewer_options
         self.layer_widget = viewer.layer_options
+
+    @observe("show_uncertainty")
+    def _toggle_uncertainty(self, event):
+        spec_viewer = self.app.get_viewer("spectrum-viewer")

```

Comment:
I know we want to move towards using `app.get_viewer_by_id()` but this line works with both specviz and cubeviz. `app.get_viewer_by_id()` would require the `id` to be `specviz-0` or `cubeviz-3`, respectively.

Reply:
This doesn't exist for imviz, for example, but the toggle uncertainty button still shows (and actually causes imviz to crash).